src/common.h

```diff
@@ -103,8 +99,13 @@ enum class Result {
   Error,
 };
 
-#define WABT_SUCCEEDED(x) ((x) == ::wabt::Result::Ok)
-#define WABT_FAILED(x) ((x) == ::wabt::Result::Error)
+template<typename T>
+void ZeroMemory(T& v) {
+  memset(&v, 0, sizeof(v));
```

Comment:
I think we should keep the static assert -- it's caught a number of bugs while I've refactored.